Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -5158,23 +5158,104 @@ namespace AssetProcessor
 
     void AssetProcessorManager::QueryAbsolutePathDependenciesRecursive(AZ::Uuid sourceUuid, SourceFilesForFingerprintingContainer& finalDependencyList, AzToolsFramework::AssetDatabase::SourceFileDependencyEntry::TypeOfDependency dependencyType)
     {
+        struct PathOrUuid
+        {
+            explicit PathOrUuid(AZ::Uuid uuid)
+                : m_uuid(uuid)
+            {
+            }
+
+            explicit PathOrUuid(AZStd::string path) : m_path(AZStd::move(path))
+            {
+
+            }
+
+            bool IsUuid() const
+            {
+                return !m_uuid.IsNull();
+            }
+
+            AZStd::string GetPath() const
+            {
+                return m_path;
+            }
+
+            AZ::Uuid GetUuid() const
+            {
+                return m_uuid;
+            }
+
+            AZStd::string ToString() const
+            {
+                if (IsUuid())
+                {
+                    return m_uuid.ToString<AZStd::string>(false, false);
+                }
+                else
+                {
+                    return m_path;
+                }
+            }
+
+            bool operator==(const PathOrUuid& rhs) const
+            {
+                return rhs.m_path == m_path && rhs.m_uuid == m_uuid;
+            }
+
+            AZStd::string m_path;
+            AZ::Uuid m_uuid;
```

Comment:
Seems like the `m_path` member type should be `AZ::IO::Path`

Reply:
Also instead of having a structure for this, you can just replace this entire struct and the Hasher struct as well with an `using PathOrUuid = AZStd::variant<AZ::IO::Path, AZ::Uuid>;`

It specializaes AZStd::hash as well so it can be used in hash containers out of the box.

Reply:
This is just temporary, I have another PR that is going to overhaul more of the source dependency system which introduces a proper PathOrUuid class.  Its hooked into the sqlite stuff to be able to serialize in and out to the db, allowing queries to be written which accept a PathOrUuid.  I'm not sure how well using a variant would play with this, because the UUID needs to be written to the db with no dashes, brackets or spaces (the same format generated by sqlite's hex function).  I can investigate it though.

So I could probably replace this temporary usage with the variant, but I don't think it matters much, the other PR is already posted, I just wanted to get this fix in quick while waiting on that larger PR.